test/index.js

```diff
@@ -1,701 +1,813 @@
-  return {
-    // @ts-expect-error The connection is missing here, which is ok for testing.
-    console: {
-      error: spy(),
-      info: spy(),
-      log: spy(),
-      warn: spy()
-    },
-    listen: spy(),
-    onInitialize: spy(),
-    onDidChangeConfiguration: spy(),
-    onDidChangeWatchedFiles: spy(),
-    onCodeAction: spy(),
-    onDocumentFormatting: spy(),
-    sendDiagnostics: stub()
-  }
-}
-
-/**
- * @param {string} uri
- * @param {string} text
- * @param {UnifiedTestPluginOptions} [pluginOptions]
- * @param {Processor} [processor]
- * @param {string} pluginName
- * @returns {Promise<import('vscode-languageserver').PublishDiagnosticsParams>}
- */
-function getDiagnostic(
-  uri,
-  text,
-  pluginOptions,
-  processor,
-  pluginName = './test/test-plugin.js'
-) {
-  const connection = createMockConnection()
-  const documents = new TextDocuments(TextDocument)
-  const diagnosticsPromise = new Promise((resolve) => {
-    const sendDiagnostics = /** @type import('sinon').SinonStub */ (
-      connection.sendDiagnostics
-    )
-    sendDiagnostics.callsFake(resolve)
-  })
-  const onDidChangeContent = spy()
-  Object.defineProperty(documents, 'onDidChangeContent', {
-    value: onDidChangeContent
-  })
-
-  configureUnifiedLanguageServer(connection, documents, {
-    plugins: [[pluginName, pluginOptions]],
-    processor
-  })
-
-  onDidChangeContent.firstCall.firstArg({
-    document: TextDocument.create(uri, 'text', 0, text)
-  })
-
-  return diagnosticsPromise
-}
 
-test('onInitialize', (t) => {
-  const connection = createMockConnection()
-  const documents = new TextDocuments(TextDocument)
+import * as exports from 'unified-language-server'
 
-  configureUnifiedLanguageServer(connection, documents, {})
+const sleep = promisify(setTimeout)
 
-  const initialize = /** @type import('sinon').SinonSpy */ (
-    connection.onInitialize
-  ).firstCall.firstArg
-  const result = initialize()
+const delay = process.platform === 'win32' ? 600 : 300
+const timeout = 10_000
 
-  t.deepEquals(result, {
-    capabilities: {
-      textDocumentSync: TextDocumentSyncKind.Full,
-      documentFormattingProvider: true,
-      codeActionProvider: {
-        codeActionKinds: [CodeActionKind.QuickFix],
-        resolveProvider: true
-      }
-    }
-  })
+test('exports', (t) => {
+  t.equal(typeof exports.createUnifiedLanguageServer, 'function')
 
   t.end()
 })
 
-test('Custom processor', async (t) => {
-  const uri = String(pathToFileURL('test.md'))
-  const diagnostics = await getDiagnostic(
-    uri,
-    'test',
-    undefined,
-    unified().use(
-      /** @type {Plugin} */ (
-        () => (ast, file) => {
-          file.message('custom processor')
-        }
-      )
-    )
+test('`initialize`', async (t) => {
+  await fs.writeFile(
+    new URL('lsp.js', import.meta.url),
+    `import {createUnifiedLanguageServer} from '../index.js'
+    createUnifiedLanguageServer({plugins: ['remark-parse', 'remark-stringify']})`
   )
```

Comment:
Instead of writing to disk in the tests, maybe the files can just be created and committed? I.e. this file could be committed as `test/fixtures/lsp-initialize.js`, then just be used in this test. as it’s currently used.